packages/data/src/factory.js

```diff
@@ -79,3 +79,24 @@ export function createRegistryControl( registryControl ) {
 
 	return registryControl;
 }
+
+/**
+ * Creates a store definition to pass to store API methods.
+ *
+ * @example
+ * ```js
+ * const storeDefinition = createStoreDefinition( 'my-shop' );
+ * ```
+ *
+ * @param {string} storeName Unique namespace identifier for the store.
+ *
+ * @return {Object} Store definition object.
```

Comment:
I can add a more precise JSDoc type once we agree on the API :)

Reply:
The current version works fine for me, Let's make the function unstable though because I might revisit it to create several versions of it (one of atomic, one for namespace and a generic one).

Reply:
Good point, let's do it 👍

Reply:
Done in https://github.com/WordPress/gutenberg/pull/26655/commits/ede926967b73b9b9fb49d4d417e27ffa323e66b8.